### DISCUSSION TITLE
ESP32 blocked main loop

### DISCUSSION THREAD

**Discussion by sude22** [bertmelis/espMqttClient#75] [General]
Hello Bert,
maybe you can help me with a comprehension question?
I am using an ESP32 for a project and your library for the WiFi MQTT part.
It is important for me that the loop with the control parts always goes through undisturbed and does not block.
Therefore I thought if I initialize the espMqttClient(1, 0) the WiFi and MQTT part would be executed on core 0 and in case of connection problems the main loop would not be blocked.
I'm not 100% sure yet, but in serial debug it looks like in case of problems, before MQTT disconnect, the loop is blocked.

I'm trying to collect more detailed info.

**Comment by bertmelis** [comment]
Which version are you using? The latest release or main branch?

If you're using the latest release, your code would have the client create its own task and stay out of the way of any user tasks (like the Arduino loop). So yes, the client's loop does block but it shouldn't hinder you.

PS Blocking is practically only when creating the TCP (and TLS!) connection.

**Reply by sude22** [reply]
I use latest release. Thanks for the answer. As I said, I'm looking for a way to reproduce it.
But the espMqttClient(1, 0) should be OK, is it?

**Reply by bertmelis** [reply]
Yes, that way the mqtt client has its own task and will not interfere with nor block your code.

**Reply by sude22** [reply]
OK, I created a test project with bare minimum code for MqttWifi, and with that the problem does not show up. I guess it's something in my other code. But thank you for your time and please keep up the great work.
